storybook/cypress/e2e/document-component.cy.ts

```diff
@@ -83,23 +95,92 @@ describe("Document Component", () => {
     it(`matches image snapshot ${state} - table-of-contents`, () => {
       const wijzigactie = state === "default" ? null : state;
 
-      cy.visit("http://localhost:45000/iframe.html?id=core-document-component--default")
-        .get("dso-document-component.hydrated")
+      cy.get("@document-component")
         .invoke("prop", "wijzigactie", wijzigactie)
         .invoke("prop", "filtered", false)
         .invoke("prop", "mode", "table-of-contents");
 
       if (wijzigactie) {
-        cy.get("dso-document-component.hydrated").should("have.attr", "wijzigactie", wijzigactie);
+        cy.get("@document-component").should("have.attr", "wijzigactie", wijzigactie);
       } else {
-        cy.get("dso-document-component.hydrated").should("not.have.attr", "wijzigactie");
+        cy.get("@document-component").should("not.have.attr", "wijzigactie");
       }
 
-      cy.get("dso-document-component.hydrated")
-        .should("have.attr", "mode", "table-of-contents")
-        .and("not.have.attr", "filtered");
+      cy.get("@document-component").should("have.attr", "mode", "table-of-contents").and("not.have.attr", "filtered");
 
-      cy.get("dso-document-component.hydrated").matchImageSnapshot();
+      cy.get("@document-component").matchImageSnapshot();
     });
   }
+
+  it("shows a gereserveerd alert when gereserveerd prop is set with XML", () => {
+    setProps({ open: true, gereserveerd: `<Gereserveerd></Gereserveerd>`, vervallen: "" });
+
+    cy.get("@document-component").shadow().find(".heading-element dso-label").should("have.text", "gereserveerd");
+
+    expectAlert("Dit onderdeel is gereserveerd voor toekomstige toevoeging.");
+    cy.get("@document-component").matchImageSnapshot(`${Cypress.currentTest.title} -- gereserveerd`);
+  });
+
+  it("shows a gereserveerd label with status verwijder when gereserveerd prop is set with verwijder XML", () => {
+    setProps({ open: true, gereserveerd: `<Gereserveerd wijzigactie='verwijder'></Gereserveerd>`, vervallen: "" });
+
+    cy.get("@document-component")
+      .shadow()
+      .find(".heading-element dso-label")
+      .should("have.text", "gereserveerd")
+      .should("have.attr", "status", "verwijderd");
+
+    expectAlert();
+    cy.get("@document-component").matchImageSnapshot(`${Cypress.currentTest.title} -- gereserveerd verwijderd`);
+  });
+
+  it("shows a gereserveerd label with a verwijderd status and a vervallen label with toegevoegd status when gereserveerd and vervallen props are set", () => {
```

Comment:
Deze suggestie had je gemist:

```suggestion
  it("shows gereserveerd label with status verwijder when gereserveerd prop is set with wijzigactie='verwijder'", () => {
```